### PR TITLE
Bump Go to 1.18

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
Along side -> https://github.com/kubernetes/test-infra/pull/25652

**Comment by ArangoGutierrez** [comment]
Depends on https://github.com/kubernetes/test-infra/pull/25652

File: go.mod [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
 require (
 	github.com/onsi/ginkgo v1.16.5
 	github.com/onsi/gomega v1.18.1
-	k8s.io/api v0.23.3
-	k8s.io/apimachinery v0.23.3
-	k8s.io/client-go v0.23.3
-	k8s.io/component-base v0.23.3
+	k8s.io/api v0.23.4
+	k8s.io/apimachinery v0.23.4
+	k8s.io/client-go v0.23.4
```
[the remark this answers is not in the document]
Comment (ArangoGutierrez) [reply go.mod:10]: will do

**Review by marquiz — APPROVED** [review]
👍 
/lgtm

Let's wait for https://github.com/kubernetes/test-infra/pull/25652 to get merged
/hold

**Comment by ArangoGutierrez** [comment]
I think we need to also update go.mod to fix the tests

**Comment by marquiz** [comment]
> I think we need to also update go.mod to fix the tests

#139 should be enough

**Pull request merged**